libmachine/provision/systemd.go

```diff
@@ -22,7 +22,7 @@ func NewSystemdProvisioner(osReleaseID string, d drivers.Driver) SystemdProvisio
 		GenericProvisioner{
 			SSHCommander:      GenericSSHCommander{Driver: d},
 			DockerOptionsDir:  "/etc/docker",
-			DaemonOptionsFile: "/etc/systemd/system/docker.service",
+			DaemonOptionsFile: "/etc/systemd/system/docker.service.d/machine.conf",
```

Comment:
I can probably run the tests on a Redhat AMI with the EC2 driver, but maybe this can/will be triggered automatically on the PR?